src/shogun/io/OpenmlFlow.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifdef HAVE_CURL
+
+#include <shogun/io/OpenmlFlow.h>
+#include "OpenmlFlow.h"
+
+
+using namespace shogun;
+
+size_t writer(char *data, size_t size, size_t nmemb, std::string* buffer_in)
+{
+	// adapted from https://stackoverflow.com/a/5780603
+	// Is there anything in the buffer?
+	if (buffer_in->empty())
+	{
+		// Append the data to the buffer
+		buffer_in->append(data, size * nmemb);
+
+		return size * nmemb;
+	}
+
+	return 0;
+}
+
+const char* OpenMLReader::xml_server = "https://www.openml.org/api/v1/xml";
+const char* OpenMLReader::json_server = "https://www.openml.org/api/v1/json";
+const char* OpenMLReader::dataset_description = "/data/{}";
+const char* OpenMLReader::list_data_qualities = "/data/qualities/list";
+const char* OpenMLReader::data_features = "/data/features/{}";
+const char* OpenMLReader::list_dataset_qualities = "/data/qualities/{}";
+const char* OpenMLReader::list_dataset_filter = "/data/list/{}";
+const char* OpenMLReader::flow_file = "/flow/{}";
+
+const std::unordered_map<std::string, std::string>
+    OpenMLReader::m_format_options = {{"xml", xml_server},
+                                      {"json", json_server}};
+const std::unordered_map<std::string, std::string>
+    OpenMLReader::m_request_options = {
+
+        {"dataset_description", dataset_description},
+        {"list_data_qualities", list_data_qualities},
+        {"data_features", data_features},
+        {"list_dataset_qualities", list_dataset_qualities},
+        {"list_dataset_filter", list_dataset_filter},
+		{"flow_file", flow_file}};
+
+OpenMLReader::OpenMLReader(const std::string& api_key) : m_api_key(api_key)
+{
+}
+
+void OpenMLReader::post(const std::string& request, const std::string& data)
+{
+}
+
+void OpenMLReader::openml_curl_request_helper(const std::string& url)
+{
+	CURL* curl_handle = nullptr;
+
+	curl_handle = curl_easy_init();
+
+	if (!curl_handle)
+	{
+		SG_SERROR("Failed to initialise curl handle.")
```

Comment:
maybe some infos on what happened?

Reply:
I am not sure what the exact error would be... From the docs: "If this function [curl_easy_init] returns NULL, something went wrong and you cannot use the other curl functions."

Reply:
Any value in using this other global init instead?
https://stackoverflow.com/questions/50987505/how-to-get-error-reason-when-curl-easy-init-fails

Reply:
ah thanks, yes, that seems helpful!